### DISCUSSION TITLE
Implementing Diffusion in NEMO 3D Data

### DISCUSSION THREAD

**Discussion by AnHyeJu** [Parcels-code/Parcels#1664] [Q&A]
## Question
Hello,

I am currently using NEMO 3D data with the AdvectionRK4_3D kernel. I would like to apply diffusion as well. I noticed in the documentation that three-dimensional advection-diffusion is not yet implemented in Parcels, but I saw this issue: https://github.com/OceanParcels/parcels/issues/858. I was wondering if there is any update on this or if there is a way to  implement diffusion with 3D velocity data using the current predefined kernels?

Thank you!

**Comment by erikvansebille** [comment]
HI @AnHyeJu, thanks for askin this question. I think it depends a bit on what type of diffusion you're after. If it's isopycnal/diapycnal diffusion in the deep ocean (below the mixing layer), then I don't think there has been much progress since #858. Note however the [Reijnders et al (2022)](https://agupubs.onlinelibrary.wiley.com/doi/full/10.1029/2021MS002850) paper for a very detailed discussion on the theme

If you're after diffusion/mixing in the mixing layer, then there is e.g. the [Onink et al (2022)](https://gmd.copernicus.org/articles/15/1995/2022/) paper, that describes how to use Markov-type vertical mixing for particles in a KPP-type scheme

Hope this helps!
